src/printer.js

```diff
@@ -45,8 +45,18 @@ function printIToken(path) {
 
 function printAttribute(path, opts, print) {
   const { Name, EQUALS, STRING } = path.getValue().children;
-
-  return [Name[0].image, EQUALS[0].image, STRING[0].image];
+  
+  let attributeValue;
+  if (opts.quoteProps === "preserve") {
+    attributeValue = STRING[0].image;
+  } else {
+    if (opts.singleQuote) {
+      attributeValue = STRING[0].image.replaceAll("\"", "\'");
```

Comment:
I think we might have a problem here because you're not handling the opposite quote. For example:

```xml
<foo bar="baz ' baz">
</foo>
```

would become:

```xml
<foo bar='baz ' baz'>
</foo>
```

which is a syntax error. So you can either replace doubles with `&quot;` and singles with `&apos;`, or you can leave them alone if they have the opposing quote.

Reply:
Ack.  You are absolutely right.  I wasn't sure that that was supported in XML and thought `&quot;` and `&apos;` were the only ways that it was possible.  Will fix.